test/common/router/req_header_formatter_test.cc

```diff
@@ -0,0 +1,104 @@
+#include <string>
+
+#include "envoy/http/protocol.h"
+
+#include "common/config/rds_json.h"
+#include "common/http/access_log/access_log_formatter.h"
+#include "common/router/req_header_formatter.h"
+
+#include "test/mocks/http/mocks.h"
+#include "test/mocks/upstream/mocks.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Router {
+using testing::NiceMock;
+using testing::Return;
+using testing::ReturnRef;
+using testing::_;
+
+static envoy::api::v2::Route parseRouteFromJson(const std::string& json_string) {
+  envoy::api::v2::Route route;
+  auto json_object_ptr = Json::Factory::loadFromString(json_string);
+  Envoy::Config::RdsJson::translateRoute(*json_object_ptr, route);
+  return route;
+}
+
+TEST(RequestHeaderFormatterTest, TestFormatWithClientIpVariable) {
+  NiceMock<Envoy::Http::AccessLog::MockRequestInfo> requestInfo;
+  std::string s1 = "127.0.0.1";
+  ON_CALL(requestInfo, getDownstreamAddress()).WillByDefault(ReturnRef(s1));
+  std::string variable = "CLIENT_IP";
+  RequestHeaderFormatter requestHeaderFormatter(variable);
+  std::string formatted_string = requestHeaderFormatter.format(requestInfo);
+  EXPECT_EQ("127.0.0.1", formatted_string);
+}
+
+TEST(RequestHeaderFormatterTest, TestFormatWithProtocolVariable) {
+  NiceMock<Envoy::Http::AccessLog::MockRequestInfo> requestInfo;
+  ON_CALL(requestInfo, protocol()).WillByDefault(Return(Envoy::Http::Protocol::Http11));
+  std::string variable = "PROTOCOL";
+  RequestHeaderFormatter requestHeaderFormatter(variable);
+  std::string formatted_string = requestHeaderFormatter.format(requestInfo);
+  EXPECT_EQ("HTTP/1.1", formatted_string);
+}
+
+TEST(RequestHeaderFormatterTest, WrongVariableToFormat) {
+  NiceMock<Envoy::Http::AccessLog::MockRequestInfo> requestInfo;
+  std::string s1 = "127.0.0.1";
+  ON_CALL(requestInfo, getDownstreamAddress()).WillByDefault(ReturnRef(s1));
+  std::string variable = "INVALID_VARIABLE";
+  ASSERT_THROW(RequestHeaderFormatter requestHeaderFormatter(variable), EnvoyException);
```

Comment:
i'm sorry but i still do not see the test case where `last_occ_pos <= 1` is exercised.